### DISCUSSION TITLE
Enhancement to work with AWS MSK IAM Authentication

### DISCUSSION THREAD

**Discussion by cookie-bytes** [BEagle1984/silverback#215] [Ideas]
With the introduction of IAM Authentication in AWS MSK, as outlined in this blog, https://aws.amazon.com/blogs/big-data/amazon-msk-iam-authentication-now-supports-all-programming-languages/, would you be receptive to a pull request that incorporates this feature into Silverback?

Modifications will likely be required in both the ConfluentProducerBuilder and ConfluentConsumerBuilder, as demonstrated in the Confluent example provided here https://github.com/aws/aws-msk-iam-sasl-signer-net. https://github.com/aws/aws-msk-iam-sasl-signer-net.

**Comment by BEagle1984** [comment]
Yes, sure. That would actually be awesome if you could push a PR (as I wouldn't have the environment to test this).

**Comment by cookie-bytes** [comment]
Migration to AWS IAM has been put on the backlog for now, so we will pick this up later.
